client/core/core.go

```diff
@@ -1159,6 +1165,37 @@ func (c *Core) Trade(pw []byte, form *TradeForm) (*Order, error) {
 		return nil, fmt.Errorf("unknown DEX %s", form.DEX)
 	}
 
+	marketName, err := dex.MarketName(form.Base, form.Quote)
+	if err != nil {
+		return nil, fmt.Errorf("market name error %v", err)
+	}
+
+	dc.suspensionsMtx.RLock()
+	ts := dc.suspensions[marketName]
+	dc.suspensionsMtx.RUnlock()
+
+	dc.resumptionsMtx.RLock()
+	tr := dc.resumptions[marketName]
+	dc.resumptionsMtx.RUnlock()
+
+	// Proceed with the order if there is no trade suspension
+	// scheduled for the market.
+	if ts != nil {
+		now := encode.UnixMilliU(time.Now())
+
+		var resumptionAt uint64
+		if tr != nil {
+			resumptionAt = tr.StartEpoch * tr.EpochLen
+		}
+
+		if (now > ts.SuspendTime && resumptionAt == 0) ||
+			(now > ts.SuspendTime && now < resumptionAt) {
+			return nil, fmt.Errorf("order placement suspended due to"+
+				" trade suspension started at %v for dex %s",
+				encode.UnixTimeMilli(int64(ts.SuspendTime)), dc.acct.url)
+		}
+	}
```

Comment:
With the `time.AfterFunc` suggestion, this could be reduced to something like:

```go
// Proceed with the order if there is no trade suspension
// scheduled for the market.
if dc.suspended(marketName(form.Base, form.Quote)) {
	return nil, fmt.Errorf("suspended market")
}
```

I was thinking the current suspend status of a market could be in the `marketMap`, even though that requires a little care in `refreshMarkets` to keep the value, because you can have methods like:

```go
func (dc *dexConnection) suspended(mkt string) bool {
	dc.marketMtx.Lock()
	defer dc.marketMtx.Unlock()
	return dc.marketMap[mkt].Suspended
}

func (dc *dexConnection) suspend(mkt string) {
	dc.marketMtx.Lock()
	dc.marketMap[mkt].Suspended = true
	dc.marketMtx.Unlock()
}

func (dc *dexConnection) resume(mkt string) {
	dc.marketMtx.Lock()
	dc.marketMap[mkt].Suspended = false
	dc.marketMtx.Unlock()
}
```